yktrace/src/tir.rs

```diff
@@ -177,9 +187,19 @@ pub enum Guard {
     Integer(Place, SerU128),
     /// The local must not be a member of the specified collection of integers.
     /// This is necessary due to the "otherwise" semantics of the SwitchInt terminator in MIR.
-    OtherInteger(Place, Vec<SerU128>),
+    OtherInteger(Place, Vec<u128>),
```

Comment:
Is it just me, or does `Guard` always have a first argument of `Place`? At which point that suggests that `Guard` should be a `struct` along the lines of:

```rust
struct Guard {
    lhs: Place
    rhs: GuardKind
}

enum GuardKind {
    Dunno(Vec<u128>)
    Bool(bool)
}
```

Reply:
We could do. I'm indifferent. Yours makes sense, but mine is marginally easier to de-structure. You choose :)

Reply:
We're going to add more variants to `GuardKind` I assume? In which case the split I suggested is definitely more idiomatic. If not, then it's maybe less important.